browser/ui/webui/settings/brave_search_engines_handler.cc

```diff
@@ -23,6 +26,33 @@ namespace settings {
 namespace {
 constexpr char kBraveSearchForTorKeyword[] =
     ":search.brave4u7jddbv7cyviptqjc7jusxh72uik7zt6adtckl5f4nwy2v72qd.onion";
+
+// Put yahoo at first place.
+void SortDefaultSearchEnginesListInJP(base::Value::List& defaults) {
+  auto* local_state = g_browser_process->local_state();
+  if (!local_state || brave_l10n::GetCountryCode(local_state) != "JP") {
+    return;
+  }
+
+  auto yahoo_jp = std::find_if(defaults.begin(), defaults.end(), [](auto& val) {
+    const auto& dict = val.GetDict();
+    const std::string* url = dict.FindString("url");
+    if (!url) {
+      return false;
+    }
+
+    return GURL(*url).host_piece() ==
+           GURL(TemplateURLPrepopulateData::brave_yahoo_jp.search_url)
```

Comment:
optional - we could pull this into a variable outside the lambda so we don't recreate the URL for each entry

Reply:
Done. 👍🏼